plugins/modules/cloudwatchlogs_log_group_metric_filter.py

```diff
@@ -184,6 +215,12 @@ def main():
         metricTransformation = [camel_dict_to_snake_dict(item) for item in [originMetricTransformations]]
 
     elif state == "present":
+        if (
+            metric_transformation.get("default_value") is not None
+            and metric_transformation.get("dimensions") is not None
+        ):
+            module.fail_json(msg="default_value and dimensions are mutually exclusive.")
```

Comment:
Since the AWS Boto3 call does not fail or throw an error when both dimensions and default values are provided, can we issue a warning and proceed with the configuration, rather than causing the module to fail?

Reply:
That is possible, however personally I prefer it to fail, because if this is only a warning, then:

1. I feel that does increase the risk that this goes unnoticed for the user, and can lead to confusion on why a metric_filter would not work as expected.
2. You will run into idempotency issues, because the dimensions will never actually be configured, so on subsequent runs tasks with both dimensions and default_value will always be 'changed'.

If you still feel this should be a warning instead of a failure, let me know then I will update the PR.

Reply:
Thank you for the quick response by the way. I have updated the `Return` block

Reply:
I would like to know  what @tremble , @alinabuzachis think about throwing a warning instead of an error for the above condition.